Add, get update proposal comment.

Eb-Zeero opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/saltastroops/salt-api-2021/28)
<!-- Reviewable:end -->

Eb-Zeero committed:
Add, get update proposal comment.

hettlage reviewed (changes requested):
> Reviewed 2 of 5 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/saltastroops/salt-api-2021/28)* status: 2 of 5 files reviewed, 14 unresolved discussions (waiting on @Eb-Zeero)

___
*[saltapi/repository/proposal_repository.py, line 22 at r1](https://reviewable.io/reviews/saltastroops/salt-api-2021/28#-MjnXvHU9tmaVDrJ9pbf:-MjnXvHU9tmaVDrJ9pbg:bsai5ym) ([raw file](https://github.com/saltastroops/salt-api-2021/blob/b503a03e4d1c9064b0010df461ba350bfed8f268/saltapi/repository/proposal_repository.py#L22)):*
> ```Python
>     tonight,
> )
> from saltapi.web.schema.common import Message
> ```

The schema types shouldn't be used at the repository (or service) level; these levels should not depend on the API level.
___
*[saltapi/repository/proposal_repository.py, line 1021 at r1](https://reviewable.io/reviews/saltastroops/salt-api-2021/28#-MjnpBllDflUvfu8ZlqV:-MjnpBllDflUvfu8ZlqW:b-3bis3m) ([raw file](https://github.com/saltastroops/salt-api-2021/blob/b503a03e4d1c9064b0010df461ba350bfed8f268/saltapi/repository/proposal_repository.py#L1021)):*
> ```Python
>         return [ObservationComment(**dict(row)) for row in result]
> 
>     def add_observation_comment(self, proposal_code: str, comment: str, user: User) -> Message:
> ```

I'd suggest to call the `user` parameter `author`.
___
*[saltapi/repository/proposal_repository.py, line 1042 at r1](https://reviewable.io/reviews/saltastroops/salt-api-2021/28#-Mjnam-R0xRxwj_JXY3Z:-Mjnam-R0xRxwj_JXY3_:bhud9q1) ([raw file](https://github.com/saltastroops/salt-api-2021/blob/b503a03e4d1c9064b0010df461ba350bfed8f268/saltapi/repository/proposal_repository.py#L1042)):*
> ```Python
>                 "proposal_code": proposal_code,
>                 "username": user.username,
>                 "comment": comment}
> ```

date missing
___
*[saltapi/repository/proposal_repository.py, line 1044 at r1](https://reviewable.io/reviews/saltastroops/salt-api-2021/28#-MjnbGVEAas_uaXqyRIp:-MjnbGVEAas_uaXqyRIq:b-umrog1) ([raw file](https://github.com/saltastroops/salt-api-2021/blob/b503a03e4d1c9064b0010df461ba350bfed8f268/saltapi/repository/proposal_repository.py#L1044)):*
> ```Python
>                 "comment": comment}
>         )
>         return Message(message="Comment added successfully.")
> ```

This function shouldn't return anything.
___
*[saltapi/repository/proposal_repository.py, line 1046 at r1](https://reviewable.io/reviews/saltastroops/salt-api-2021/28#-MjndbpK0QCHGBgAmDAH:-MjndbpK0QCHGBgAmDAI:b-lp4tla) ([raw file](https://github.com/saltastroops/salt-api-2021/blob/b503a03e4d1c9064b0010df461ba350bfed8f268/saltapi/repository/proposal_repository.py#L1046)):*
> ```Python
>         return Message(message="Comment added successfully.")
> 
>     def update_observation_comment(self, comment_id: int, comment: str) -> Message:
> ```

I'm not entirely convinced that updating a comment without even a hint that it has been updated is a good idea. From this perspective, it might be better not to include an update method for now.
___
*[saltapi/repository/proposal_repository.py, line 1057 at r1](https://reviewable.io/reviews/saltastroops/salt-api-2021/28#-MjnbRxB3xmIMZLSKrTa:-MjnbRxB3xmIMZLSKrTb:b-umrog1) ([raw file](https://github.com/saltastroops/salt-api-2021/blob/b503a03e4d1c9064b0010df461ba350bfed8f268/saltapi/repository/proposal_repository.py#L1057)):*
> ```Python
>             stmt, {"comment_id": comment_id, "comment": comment}
>         )
>         return Message(message="Updated successfully.")
> ```

This function shouldn't return anything.
___
*[saltapi/repository/proposal_repository.py, line 1057 at r1](https://reviewable.io/reviews/saltastroops/salt-api-2021/28#-MjndJV-AKeUmlfYBuJU:-MjndJV-AKeUmlfYBuJV:bh7id54) ([raw file](https://github.com/saltastroops/salt-api-2021/blob/b503a03e4d1c9064b0010df461ba350bfed8f268/saltapi/repository/proposal_repository.py#L1057)):*
> ```Python
>             stmt, {"comment_id": comment_id, "comment": comment}
>         )
>         return Message(message="Updated successfully.")
> ```

The function should raise a NotFoundError if the comment id does not exist
___
*[saltapi/service/proposal_service.py, line 6 at r1](https://reviewable.io/reviews/saltastroops/salt-api-2021/28#-MjnfAS_CtwemIah5T_5:-MjnfAS_CtwemIah5T_6:b-x79fpa) ([raw file](https://github.com/saltastroops/salt-api-2021/blob/b503a03e4d1c9064b0010df461ba350bfed8f268/saltapi/service/proposal_service.py#L6)):*
> ```Python
> from saltapi.service.proposal import Proposal, ProposalListItem
> from saltapi.service.user import User
> from saltapi.web.schema.common import Message
> ```

API schema types shouldn't be used on the service level
___
*[saltapi/service/proposal_service.py, line 21 at r1](https://reviewable.io/reviews/saltastroops/salt-api-2021/28#-MjneCeK6OaTjfcN5kGs:-MjneCeK6OaTjfcN5kGt:b3f5lnm) ([raw file](https://github.com/saltastroops/salt-api-2021/blob/b503a03e4d1c9064b0010df461ba350bfed8f268/saltapi/service/proposal_service.py#L21)):*
> ```Python
> 
>     def get_observation_comments(self, proposal_code: str) -> List[ObservationComment]:
>         return self.repository.get_observation_comments(proposal_code)
> ```

Permission check missing
___
*[saltapi/service/proposal_service.py, line 25 at r1](https://reviewable.io/reviews/saltastroops/salt-api-2021/28#-Mjnemr44SLzxTbstpoO:-Mjnemr44SLzxTbstpoP:b-e2rxig) ([raw file](https://github.com/saltastroops/salt-api-2021/blob/b503a03e4d1c9064b0010df461ba350bfed8f268/saltapi/service/proposal_service.py#L25)):*
> ```Python
>     def add_observation_comment(self, proposal_code: str, comment: str, user: User) -> Message:
>         # TODO only SA's, SO's and investigators can add a comment
>         return self.repository.add_observation_comment(proposal_code, comment, user)
> ```

This method shouldn't return anything
___
*[saltapi/service/proposal_service.py, line 27 at r1](https://reviewable.io/reviews/saltastroops/salt-api-2021/28#-Mjnf-Rg4qZoinPj84Ed:-Mjnf-Rg4qZoinPj84Ee:b-e2rxig) ([raw file](https://github.com/saltastroops/salt-api-2021/blob/b503a03e4d1c9064b0010df461ba350bfed8f268/saltapi/service/proposal_service.py#L27)):*
> ```Python
>         return self.repository.add_observation_comment(proposal_code, comment, user)
> 
>     def update_observation_comment(self, comment_id: int, comment: str, user: User) -> Message:
> ```

This method shouldn't return anything
___
*[saltapi/web/api/proposals.py, line 330 at r1](https://reviewable.io/reviews/saltastroops/salt-api-2021/28#-MjnnRq7DYIGTKBn4bwL:-MjnnRq7DYIGTKBn4bwM:bkaxjsw) ([raw file](https://github.com/saltastroops/salt-api-2021/blob/b503a03e4d1c9064b0010df461ba350bfed8f268/saltapi/web/api/proposals.py#L330)):*
> ```Python
>     ),
>     comment: str = Body(..., title="Comment", description="Text of the comment."),
>     user: User = None    # TODO needs to get and authorised user.
> ```

There is a function for this already:

```
user: User = Depends(get_current_user)
```
___
*[saltapi/web/api/proposals.py, line 331 at r1](https://reviewable.io/reviews/saltastroops/salt-api-2021/28#-MjnoLTn3NiD-c5YroJn:-MjnoLTn3NiD-c5YroJo:b-domx4x) ([raw file](https://github.com/saltastroops/salt-api-2021/blob/b503a03e4d1c9064b0010df461ba350bfed8f268/saltapi/web/api/proposals.py#L331)):*
> ```Python
>     comment: str = Body(..., title="Comment", description="Text of the comment."),
>     user: User = None    # TODO needs to get and authorised user.
> ) -> Message:
> ```

I would suggest that the endpoint should return the created comment.
___
*[saltapi/web/api/proposals.py, line 358 at r1](https://reviewable.io/reviews/saltastroops/salt-api-2021/28#-MjnobRMCZYj8TDptkPn:-MjnobRMCZYj8TDptkPo:b-domx4x) ([raw file](https://github.com/saltastroops/salt-api-2021/blob/b503a03e4d1c9064b0010df461ba350bfed8f268/saltapi/web/api/proposals.py#L358)):*
> ```Python
>                             description="Text of the comment."),
>         user: User = None    # TODO needs to get and authorised user.
> ) -> Message:
> ```

I would suggest that the endpoint should return the created comment.


<!-- Sent from Reviewable.io -->

Eb-Zeero reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/saltastroops/salt-api-2021/28)* status: 2 of 5 files reviewed, 14 unresolved discussions (waiting on @hettlage)

___
*[saltapi/repository/proposal_repository.py, line 22 at r1](https://reviewable.io/reviews/saltastroops/salt-api-2021/28#-MjnXvHU9tmaVDrJ9pbf:-Mk1ib0oElBMvsz-1UgB:b-896fix) ([raw file](https://github.com/saltastroops/salt-api-2021/blob/b503a03e4d1c9064b0010df461ba350bfed8f268/saltapi/repository/proposal_repository.py#L22)):*
<details><summary><i>Previously, hettlage wrote…</i></summary><blockquote>

The schema types shouldn't be used at the repository (or service) level; these levels should not depend on the API level.
</blockquote></details>

Done.
___
*[saltapi/repository/proposal_repository.py, line 1021 at r1](https://reviewable.io/reviews/saltastroops/salt-api-2021/28#-MjnpBllDflUvfu8ZlqV:-Mk1inttBhuDYhghSzXH:b-896fix) ([raw file](https://github.com/saltastroops/salt-api-2021/blob/b503a03e4d1c9064b0010df461ba350bfed8f268/saltapi/repository/proposal_repository.py#L1021)):*
<details><summary><i>Previously, hettlage wrote…</i></summary><blockquote>

I'd suggest to call the `user` parameter `author`.
</blockquote></details>

Done.
___
*[saltapi/repository/proposal_repository.py, line 1042 at r1](https://reviewable.io/reviews/saltastroops/salt-api-2021/28#-Mjnam-R0xRxwj_JXY3Z:-Mk1ipQR65nrVun8yTEN:b-9755fa) ([raw file](https://github.com/saltastroops/salt-api-2021/blob/b503a03e4d1c9064b0010df461ba350bfed8f268/saltapi/repository/proposal_repository.py#L1042)):*
<details><summary><i>Previously, hettlage wrote…</i></summary><blockquote>

date missing
</blockquote></details>

Thank you.
___
*[saltapi/repository/proposal_repository.py, line 1044 at r1](https://reviewable.io/reviews/saltastroops/salt-api-2021/28#-MjnbGVEAas_uaXqyRIp:-Mk1jdexAWOVyalFsF6-:b-896fix) ([raw file](https://github.com/saltastroops/salt-api-2021/blob/b503a03e4d1c9064b0010df461ba350bfed8f268/saltapi/repository/proposal_repository.py#L1044)):*
<details><summary><i>Previously, hettlage wrote…</i></summary><blockquote>

This function shouldn't return anything.
</blockquote></details>

Done.
___
*[saltapi/repository/proposal_repository.py, line 1046 at r1](https://reviewable.io/reviews/saltastroops/salt-api-2021/28#-MjndbpK0QCHGBgAmDAH:-Mk1jiap-NWKRIxifwfV:ba3dfob) ([raw file](https://github.com/saltastroops/salt-api-2021/blob/b503a03e4d1c9064b0010df461ba350bfed8f268/saltapi/repository/proposal_repository.py#L1046)):*
<details><summary><i>Previously, hettlage wrote…</i></summary><blockquote>

I'm not entirely convinced that updating a comment without even a hint that it has been updated is a good idea. From this perspective, it might be better not to include an update method for now.
</blockquote></details>

Removed
___
*[saltapi/repository/proposal_repository.py, line 1057 at r1](https://reviewable.io/reviews/saltastroops/salt-api-2021/28#-MjnbRxB3xmIMZLSKrTa:-Mk1jx3d-XBO0pHRFlI5:b-hfpv0l) ([raw file](https://github.com/saltastroops/salt-api-2021/blob/b503a03e4d1c9064b0010df461ba350bfed8f268/saltapi/repository/proposal_repository.py#L1057)):*
<details><summary><i>Previously, hettlage wrote…</i></summary><blockquote>

This function shouldn't return anything.
</blockquote></details>

removed
___
*[saltapi/repository/proposal_repository.py, line 1057 at r1](https://reviewable.io/reviews/saltastroops/salt-api-2021/28#-MjndJV-AKeUmlfYBuJU:-Mk1jzTJ7mKzM7InmhRW:b-hfpv0l) ([raw file](https://github.com/saltastroops/salt-api-2021/blob/b503a03e4d1c9064b0010df461ba350bfed8f268/saltapi/repository/proposal_repository.py#L1057)):*
<details><summary><i>Previously, hettlage wrote…</i></summary><blockquote>

The function should raise a NotFoundError if the comment id does not exist
</blockquote></details>

removed
___
*[saltapi/service/proposal_service.py, line 6 at r1](https://reviewable.io/reviews/saltastroops/salt-api-2021/28#-MjnfAS_CtwemIah5T_5:-Mk1k1z03wxTrjU8GU-L:ba3dfob) ([raw file](https://github.com/saltastroops/salt-api-2021/blob/b503a03e4d1c9064b0010df461ba350bfed8f268/saltapi/service/proposal_service.py#L6)):*
<details><summary><i>Previously, hettlage wrote…</i></summary><blockquote>

API schema types shouldn't be used on the service level
</blockquote></details>

Removed
___
*[saltapi/service/proposal_service.py, line 21 at r1](https://reviewable.io/reviews/saltastroops/salt-api-2021/28#-MjneCeK6OaTjfcN5kGs:-Mk1k5ua77gbB9NYjXte:b-j08cax) ([raw file](https://github.com/saltastroops/salt-api-2021/blob/b503a03e4d1c9064b0010df461ba350bfed8f268/saltapi/service/proposal_service.py#L21)):*
<details><summary><i>Previously, hettlage wrote…</i></summary><blockquote>

Permission check missing
</blockquote></details>

I didn't do this becouse I feel like the user service is not doing any thing.
and alll the questions I need to ask user repository.
while all the proposal services will depends the user service.
Wee need to discuss this.
___
*[saltapi/service/proposal_service.py, line 25 at r1](https://reviewable.io/reviews/saltastroops/salt-api-2021/28#-Mjnemr44SLzxTbstpoO:-Mk1kASBDWTeibhqTaap:bb7f2mf) ([raw file](https://github.com/saltastroops/salt-api-2021/blob/b503a03e4d1c9064b0010df461ba350bfed8f268/saltapi/service/proposal_service.py#L25)):*
<details><summary><i>Previously, hettlage wrote…</i></summary><blockquote>

This method shouldn't return anything
</blockquote></details>

done
___
*[saltapi/service/proposal_service.py, line 27 at r1](https://reviewable.io/reviews/saltastroops/salt-api-2021/28#-Mjnf-Rg4qZoinPj84Ed:-Mk1kCer-gp4XvrGxSon:b-896fix) ([raw file](https://github.com/saltastroops/salt-api-2021/blob/b503a03e4d1c9064b0010df461ba350bfed8f268/saltapi/service/proposal_service.py#L27)):*
<details><summary><i>Previously, hettlage wrote…</i></summary><blockquote>

This method shouldn't return anything
</blockquote></details>

Done.
___
*[saltapi/web/api/proposals.py, line 330 at r1](https://reviewable.io/reviews/saltastroops/salt-api-2021/28#-MjnnRq7DYIGTKBn4bwL:-Mk1kwEc7smDjPWT-R9M:b-896fix) ([raw file](https://github.com/saltastroops/salt-api-2021/blob/b503a03e4d1c9064b0010df461ba350bfed8f268/saltapi/web/api/proposals.py#L330)):*
<details><summary><i>Previously, hettlage wrote…</i></summary><blockquote>

There is a function for this already:

```
user: User = Depends(get_current_user)
```
</blockquote></details>

Done.
___
*[saltapi/web/api/proposals.py, line 331 at r1](https://reviewable.io/reviews/saltastroops/salt-api-2021/28#-MjnoLTn3NiD-c5YroJn:-Mk1kaA_AWVhxlupxjeK:bag84eo) ([raw file](https://github.com/saltastroops/salt-api-2021/blob/b503a03e4d1c9064b0010df461ba350bfed8f268/saltapi/web/api/proposals.py#L331)):*
<details><summary><i>Previously, hettlage wrote…</i></summary><blockquote>

I would suggest that the endpoint should return the created comment.
</blockquote></details>

It will have no meaning to any one recieving it.
___
*[saltapi/web/api/proposals.py, line 358 at r1](https://reviewable.io/reviews/saltastroops/salt-api-2021/28#-MjnobRMCZYj8TDptkPn:-Mk1kpYv9P9DhNVRXIeo:b-hfpv0l) ([raw file](https://github.com/saltastroops/salt-api-2021/blob/b503a03e4d1c9064b0010df461ba350bfed8f268/saltapi/web/api/proposals.py#L358)):*
<details><summary><i>Previously, hettlage wrote…</i></summary><blockquote>

I would suggest that the endpoint should return the created comment.
</blockquote></details>

removed


<!-- Sent from Reviewable.io -->

Eb-Zeero committed:
review fix

Eb-Zeero committed:
fix broken query.
fix params issue.
reason and original exposure time are noot always available in the database.

Eb-Zeero committed:
Merge branch 'development' of https://github.com/saltastroops/salt-api-2021 into observation-comments

 Conflicts:
	saltapi/repository/proposal_repository.py
	saltapi/repository/user_repository.py
	saltapi/service/proposal_service.py
	saltapi/web/api/proposals.py

hettlage reviewed (changes requested):
> Reviewed 1 of 9 files at r2, 12 of 12 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/saltastroops/salt-api-2021/28)* status: all files reviewed, 11 unresolved discussions (waiting on @Eb-Zeero)

___
*[docs/development.md, line 355 at r3](https://reviewable.io/reviews/saltastroops/salt-api-2021/28#-MlRAy6R2GsR1dq9wNmu:-MlRAy6R2GsR1dq9wNmv:bd3zywy) ([raw file](https://github.com/saltastroops/salt-api-2021/blob/873992b79949b3a20d08fd7128e16f0cf76894f6/docs/development.md#L355)):*
> ```Markdown
>     Permission to view a proposal.
> 
>     A author may view a proposal if at least one of the following is true.
> ```

I think "user" is the correct word here.
___
*[saltapi/settings.py, line 19 at r3](https://reviewable.io/reviews/saltastroops/salt-api-2021/28#-MlRB92a61EBrwCUEOAn:-MlRB92a61EBrwCUEOAo:bd3zywy) ([raw file](https://github.com/saltastroops/salt-api-2021/blob/873992b79949b3a20d08fd7128e16f0cf76894f6/saltapi/settings.py#L19)):*
> ```Python
> 
>     # DSN for the SALT Science Database, in a format understood by SQL Alchemy
>     # Example: mysql+pymysql://author:password@database.server:3306/sdb
> ```

I think "user" is the correct word here.
___
*[saltapi/repository/proposal_repository.py, line 1021 at r1](https://reviewable.io/reviews/saltastroops/salt-api-2021/28#-MjnpBllDflUvfu8ZlqV:-MlRCjGC1dNDePPWKj9i:b-5y78e4) ([raw file](https://github.com/saltastroops/salt-api-2021/blob/b503a03e4d1c9064b0010df461ba350bfed8f268/saltapi/repository/proposal_repository.py#L1021)):*
<details><summary><i>Previously, Eb-Zeero (Nhlavutelo Macebele) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

My comment only referred to this method. Everywhere else the "user" refers to the currently logged in user, but here the meaning is "author of the comment".
___
*[saltapi/repository/proposal_repository.py, line 1126 at r3](https://reviewable.io/reviews/saltastroops/salt-api-2021/28#-MlRCBXc7nrltARZJZd3:-MlRCBXc7nrltARZJZd4:b-ys8q7a) ([raw file](https://github.com/saltastroops/salt-api-2021/blob/873992b79949b3a20d08fd7128e16f0cf76894f6/saltapi/repository/proposal_repository.py#L1126)):*
> ```Python
>             }
>         )
>         self.connection.commit()
> ```

It shouldn't be the task of this method to commit the transaction; this should happen at the API level.
___
*[saltapi/service/permission_service.py, line 16 at r3](https://reviewable.io/reviews/saltastroops/salt-api-2021/28#-MlRCSQI2URpAibbzf9B:-MlRCSQI2URpAibbzf9C:bd3zywy) ([raw file](https://github.com/saltastroops/salt-api-2021/blob/873992b79949b3a20d08fd7128e16f0cf76894f6/saltapi/service/permission_service.py#L16)):*
> ```Python
>     def may_view_proposal(self, user: User, proposal_code: ProposalCode) -> bool:
>         """
>         Check whether the author may view a proposal.
> ```

I think "user" is the correct word here.
___
*[saltapi/service/proposal_service.py, line 21 at r1](https://reviewable.io/reviews/saltastroops/salt-api-2021/28#-MjneCeK6OaTjfcN5kGs:-MlRE1mTASC8-6II9dkQ:bvqpk7p) ([raw file](https://github.com/saltastroops/salt-api-2021/blob/b503a03e4d1c9064b0010df461ba350bfed8f268/saltapi/service/proposal_service.py#L21)):*
<details><summary><i>Previously, Eb-Zeero (Nhlavutelo Macebele) wrote…</i></summary><blockquote>

I didn't do this becouse I feel like the user service is not doing any thing.
and alll the questions I need to ask user repository.
while all the proposal services will depends the user service.
Wee need to discuss this.
</blockquote></details>

Actually, permission checking happens at the API level, so please ignore my comment.
___
*[saltapi/web/api/proposals.py, line 331 at r1](https://reviewable.io/reviews/saltastroops/salt-api-2021/28#-MjnoLTn3NiD-c5YroJn:-MlRFZ_96Zjy-o7sADf1:b-ywojxr) ([raw file](https://github.com/saltastroops/salt-api-2021/blob/b503a03e4d1c9064b0010df461ba350bfed8f268/saltapi/web/api/proposals.py#L331)):*
<details><summary><i>Previously, Eb-Zeero (Nhlavutelo Macebele) wrote…</i></summary><blockquote>

It will have no meaning to any one recieving it.
</blockquote></details>

It should have meaning, as it follows the convention that you return created resources (see https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/201 -- interestingly that page suggests that there should be a Location header as well) . If anything, it reassures the user that all went well, especially if you explicitly query the database for getting the created resource. This is not trivial: For example, some characters might not be stored properly, or the date might be wrong.

Actually, speaking of created resources, the status code should be 201.
___
*[saltapi/web/api/proposals.py, line 348 at r3](https://reviewable.io/reviews/saltastroops/salt-api-2021/28#-MlRENP01Ijii4X3FeC1:-MlRENP01Ijii4X3FeC2:b2fwne) ([raw file](https://github.com/saltastroops/salt-api-2021/blob/873992b79949b3a20d08fd7128e16f0cf76894f6/saltapi/web/api/proposals.py#L348)):*
> ```Python
> ) -> List[ObservationComment]:
>     with UnitOfWork() as unit_of_work:
>         proposal_repository = ProposalRepository(unit_of_work.connection)
> ```

Permission checking is missing
___
*[saltapi/web/api/proposals.py, line 364 at r3](https://reviewable.io/reviews/saltastroops/salt-api-2021/28#-MlREcFt-vpVO5S3ZsRz:-MlREcFt-vpVO5S3ZsS-:b-slw1ba) ([raw file](https://github.com/saltastroops/salt-api-2021/blob/873992b79949b3a20d08fd7128e16f0cf76894f6/saltapi/web/api/proposals.py#L364)):*
> ```Python
>         description="Proposal code of the proposal for which an observation comment is added.",
>     ),
>     comment: Dict[str, str] = Body(..., title="Comment", description="Text of the comment."),
> ```

You are right this should not be a str. But then it should be a Pydantic type, so that it is automatically validated. 
___
*[saltapi/web/schema/proposal.py, line 229 at r3](https://reviewable.io/reviews/saltastroops/salt-api-2021/28#-MlRHvlW5vJpbWBXE3Fe:-MlRHvlW5vJpbWBXE3Ff:b-j73j2p) ([raw file](https://github.com/saltastroops/salt-api-2021/blob/7e3c255487acba48bb4276977d502afb3b2a640b/saltapi/web/schema/proposal.py#L229)):*
> ```Python
>                 "author": "Sipho Mangana",
>                 "comment": "Please check the position angle.",
>                 "madeAt": "2019-08-24T14:15:22Z",
> ```

Good catch. And "2019-08-24T14:15:22Z" should be "2019-08-24".
___
*[tests/repository/test_user_repository.py, line 391 at r3](https://reviewable.io/reviews/saltastroops/salt-api-2021/28#-MlRInI-90w_MnpkdShf:-MlRInI-90w_MnpkdShg:bd3zywy) ([raw file](https://github.com/saltastroops/salt-api-2021/blob/873992b79949b3a20d08fd7128e16f0cf76894f6/tests/repository/test_user_repository.py#L391)):*
> ```Python
>     username = data["username"]
> 
>     # Make sure the author exists
> ```

I think "user" is the correct word here.
___
*[tests/service/test_permission_service.py, line 121 at r3](https://reviewable.io/reviews/saltastroops/salt-api-2021/28#-MlRIr-v4h5N8Kb1u2Fy:-MlRIr-v4h5N8Kb1u2Fz:bd3zywy) ([raw file](https://github.com/saltastroops/salt-api-2021/blob/873992b79949b3a20d08fd7128e16f0cf76894f6/tests/service/test_permission_service.py#L121)):*
> ```Python
> ) -> List[Tuple[str, UserRepository, bool]]:
>     """
>     Create a fake author repository for every role and return these along with the role
> ```

I think "user" is the correct word here.


<!-- Sent from Reviewable.io -->

Eb-Zeero committed:
review fix and bad refactoring fix.

Eb-Zeero reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/saltastroops/salt-api-2021/28)* status: all files reviewed, 11 unresolved discussions (waiting on @Eb-Zeero and @hettlage)

___
*[docs/development.md, line 355 at r3](https://reviewable.io/reviews/saltastroops/salt-api-2021/28#-MlRAy6R2GsR1dq9wNmu:-Mljspo9EGW8n6YWXnt1:b-896fix) ([raw file](https://github.com/saltastroops/salt-api-2021/blob/873992b79949b3a20d08fd7128e16f0cf76894f6/docs/development.md#L355)):*
<details><summary><i>Previously, hettlage wrote…</i></summary><blockquote>

I think "user" is the correct word here.
</blockquote></details>

Done.
___
*[saltapi/settings.py, line 19 at r3](https://reviewable.io/reviews/saltastroops/salt-api-2021/28#-MlRB92a61EBrwCUEOAn:-Mljt3nW-rXSszi33lEI:b-sxleio) ([raw file](https://github.com/saltastroops/salt-api-2021/blob/873992b79949b3a20d08fd7128e16f0cf76894f6/saltapi/settings.py#L19)):*
<details><summary><i>Previously, hettlage wrote…</i></summary><blockquote>

I think "user" is the correct word here.
</blockquote></details>

Sorry it is a bad renaming of a vatiable
___
*[saltapi/repository/proposal_repository.py, line 1021 at r1](https://reviewable.io/reviews/saltastroops/salt-api-2021/28#-MjnpBllDflUvfu8ZlqV:-MljtLEVD7a-I1-Sr_It:b-n9nhg4) ([raw file](https://github.com/saltastroops/salt-api-2021/blob/b503a03e4d1c9064b0010df461ba350bfed8f268/saltapi/repository/proposal_repository.py#L1021)):*
<details><summary><i>Previously, hettlage wrote…</i></summary><blockquote>

My comment only referred to this method. Everywhere else the "user" refers to the currently logged in user, but here the meaning is "author of the comment".
</blockquote></details>

I made a refactorring mistake. and I didn't notice
___
*[saltapi/repository/proposal_repository.py, line 1126 at r3](https://reviewable.io/reviews/saltastroops/salt-api-2021/28#-MlRCBXc7nrltARZJZd3:-Mlp-PJNBCTJdJkorW2l:b-896fix) ([raw file](https://github.com/saltastroops/salt-api-2021/blob/873992b79949b3a20d08fd7128e16f0cf76894f6/saltapi/repository/proposal_repository.py#L1126)):*
<details><summary><i>Previously, hettlage wrote…</i></summary><blockquote>

It shouldn't be the task of this method to commit the transaction; this should happen at the API level.
</blockquote></details>

Done.
___
*[saltapi/service/permission_service.py, line 16 at r3](https://reviewable.io/reviews/saltastroops/salt-api-2021/28#-MlRCSQI2URpAibbzf9B:-MljuyWi8QZs0JQF1Rx0:b-896fix) ([raw file](https://github.com/saltastroops/salt-api-2021/blob/873992b79949b3a20d08fd7128e16f0cf76894f6/saltapi/service/permission_service.py#L16)):*
<details><summary><i>Previously, hettlage wrote…</i></summary><blockquote>

I think "user" is the correct word here.
</blockquote></details>

Done.
___
*[saltapi/service/proposal_service.py, line 21 at r1](https://reviewable.io/reviews/saltastroops/salt-api-2021/28#-MjneCeK6OaTjfcN5kGs:-Mljv3zYDxyQL-cQ82An:b-896fix) ([raw file](https://github.com/saltastroops/salt-api-2021/blob/b503a03e4d1c9064b0010df461ba350bfed8f268/saltapi/service/proposal_service.py#L21)):*
<details><summary><i>Previously, hettlage wrote…</i></summary><blockquote>

Actually, permission checking happens at the API level, so please ignore my comment.
</blockquote></details>

Done.
___
*[saltapi/web/api/proposals.py, line 348 at r3](https://reviewable.io/reviews/saltastroops/salt-api-2021/28#-MlRENP01Ijii4X3FeC1:-Mlxlq5i4zfRM5BmUZnV:b-896fix) ([raw file](https://github.com/saltastroops/salt-api-2021/blob/873992b79949b3a20d08fd7128e16f0cf76894f6/saltapi/web/api/proposals.py#L348)):*
<details><summary><i>Previously, hettlage wrote…</i></summary><blockquote>

Permission checking is missing
</blockquote></details>

Done.
___
*[saltapi/web/api/proposals.py, line 364 at r3](https://reviewable.io/reviews/saltastroops/salt-api-2021/28#-MlREcFt-vpVO5S3ZsRz:-MljwDabBdMp2Ygfcq_5:b-896fix) ([raw file](https://github.com/saltastroops/salt-api-2021/blob/873992b79949b3a20d08fd7128e16f0cf76894f6/saltapi/web/api/proposals.py#L364)):*
<details><summary><i>Previously, hettlage wrote…</i></summary><blockquote>

You are right this should not be a str. But then it should be a Pydantic type, so that it is automatically validated. 
</blockquote></details>

Done.
___
*[saltapi/web/schema/proposal.py, line 229 at r3](https://reviewable.io/reviews/saltastroops/salt-api-2021/28#-MlRHvlW5vJpbWBXE3Fe:-Mljwt2rBTTnIF10ktxN:b-896fix) ([raw file](https://github.com/saltastroops/salt-api-2021/blob/7e3c255487acba48bb4276977d502afb3b2a640b/saltapi/web/schema/proposal.py#L229)):*
<details><summary><i>Previously, hettlage wrote…</i></summary><blockquote>

Good catch. And "2019-08-24T14:15:22Z" should be "2019-08-24".
</blockquote></details>

Done.
___
*[tests/repository/test_user_repository.py, line 391 at r3](https://reviewable.io/reviews/saltastroops/salt-api-2021/28#-MlRInI-90w_MnpkdShf:-MljwvN6EKxOOZAY4iSN:bp0i66p) ([raw file](https://github.com/saltastroops/salt-api-2021/blob/873992b79949b3a20d08fd7128e16f0cf76894f6/tests/repository/test_user_repository.py#L391)):*
<details><summary><i>Previously, hettlage wrote…</i></summary><blockquote>

I think "user" is the correct word here.
</blockquote></details>

Sorry bad refactoring.
___
*[tests/service/test_permission_service.py, line 121 at r3](https://reviewable.io/reviews/saltastroops/salt-api-2021/28#-MlRIr-v4h5N8Kb1u2Fy:-Mljx3QQ473cwAJIwl2Q:b-czpz3a) ([raw file](https://github.com/saltastroops/salt-api-2021/blob/873992b79949b3a20d08fd7128e16f0cf76894f6/tests/service/test_permission_service.py#L121)):*
<details><summary><i>Previously, hettlage wrote…</i></summary><blockquote>

I think "user" is the correct word here.
</blockquote></details>

orry bad refactoring.


<!-- Sent from Reviewable.io -->

hettlage reviewed (changes requested):
> Reviewed 9 of 9 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/saltastroops/salt-api-2021/28)* status: all files reviewed, 14 unresolved discussions (waiting on @Eb-Zeero and @hettlage)

___
*[saltapi/repository/proposal_repository.py, line 1021 at r1](https://reviewable.io/reviews/saltastroops/salt-api-2021/28#-MjnpBllDflUvfu8ZlqV:-Mm-CtwL1EO0gGa_y-e4:b-6z5zm7) ([raw file](https://github.com/saltastroops/salt-api-2021/blob/b503a03e4d1c9064b0010df461ba350bfed8f268/saltapi/repository/proposal_repository.py#L1021)):*
<details><summary><i>Previously, Eb-Zeero (Nhlavutelo Macebele) wrote…</i></summary><blockquote>

I made a refactorring mistake. and I didn't notice
</blockquote></details>

But here "author" was correct... ;-)
___
*[saltapi/repository/proposal_repository.py, line 23 at r3](https://reviewable.io/reviews/saltastroops/salt-api-2021/28#-Mm-D_hG9K8i3NhJyOEe:-Mm-D_hG9K8i3NhJyOEf:b7dmmz0) ([raw file](https://github.com/saltastroops/salt-api-2021/blob/873992b79949b3a20d08fd7128e16f0cf76894f6/saltapi/repository/proposal_repository.py#L23)):*
> ```Python
>     tonight,
> )
> from saltapi.web.schema.proposal import ObservationComment
> ```

As ObservationComment is an API level type, it shouldn't be used in a repository.
___
*[saltapi/repository/proposal_repository.py, line 1128 at r4](https://reviewable.io/reviews/saltastroops/salt-api-2021/28#-Mm-DxgR9A1rr6uZ329T:-Mm-DxgR9A1rr6uZ329U:bs9tw42) ([raw file](https://github.com/saltastroops/salt-api-2021/blob/639666ff12c60fb6ef9a2a6b00b51eb711f2e378/saltapi/repository/proposal_repository.py#L1128)):*
> ```Python
>             }
>         )
>         return ObservationComment(
> ```

Can we discuss the return value?
___
*[saltapi/service/permission_service.py, line 49 at r4](https://reviewable.io/reviews/saltastroops/salt-api-2021/28#-Mm-Er6o0BnLFBELVNy5:-Mm-Er6o0BnLFBELVNy6:b3mgte) ([raw file](https://github.com/saltastroops/salt-api-2021/blob/639666ff12c60fb6ef9a2a6b00b51eb711f2e378/saltapi/service/permission_service.py#L49)):*
> ```Python
>         Check whether the user may activate a proposal.
> 
>         This is the case if the user is any of the following:
> ```

        This is the case if the user is any of the following:

        * the Principal Investigator (and the proposal can be activated by the PI or PC)
        * the Principal Contact (and the proposal can be activated by the PI or PC)
        * a SALT Astronomer
        * an administrator
___
*[saltapi/service/permission_service.py, line 79 at r4](https://reviewable.io/reviews/saltastroops/salt-api-2021/28#-Mm-FLG32Ig-qLqRoW_5:-Mm-FLG32Ig-qLqRoW_6:b7ig492) ([raw file](https://github.com/saltastroops/salt-api-2021/blob/639666ff12c60fb6ef9a2a6b00b51eb711f2e378/saltapi/service/permission_service.py#L79)):*
> ```Python
>         This is the case if the user is any of the following:
> 
>         * a Principal Investigator
> ```

a Principal ---> the Principal
___
*[saltapi/service/permission_service.py, line 80 at r4](https://reviewable.io/reviews/saltastroops/salt-api-2021/28#-Mm-FQI8-0ARtbPywGqn:-Mm-FQI8-0ARtbPywGqo:b7ig492) ([raw file](https://github.com/saltastroops/salt-api-2021/blob/639666ff12c60fb6ef9a2a6b00b51eb711f2e378/saltapi/service/permission_service.py#L80)):*
> ```Python
> 
>         * a Principal Investigator
>         * a Principal Contact
> ```

a Principal ---> the Principal
___
*[saltapi/service/permission_service.py, line 108 at r4](https://reviewable.io/reviews/saltastroops/salt-api-2021/28#-Mm-EB3438hZdeOQ-P5W:-Mm-EB3438hZdeOQ-P5X:bp8tvfr) ([raw file](https://github.com/saltastroops/salt-api-2021/blob/639666ff12c60fb6ef9a2a6b00b51eb711f2e378/saltapi/service/permission_service.py#L108)):*
> ```Python
>         ) or self.user_repository.is_administrator(username)
> 
>     def may_add_comment(self, user: User, proposal_code: ProposalCode):
> ```

Better: may_add_observation_comment (as there are other types of comment).
___
*[saltapi/service/permission_service.py, line 114 at r4](https://reviewable.io/reviews/saltastroops/salt-api-2021/28#-Mm-FSyH2QsuKPPQ1io-:-Mm-FSyH2QsuKPPQ1io0:b16abh1) ([raw file](https://github.com/saltastroops/salt-api-2021/blob/639666ff12c60fb6ef9a2a6b00b51eb711f2e378/saltapi/service/permission_service.py#L114)):*
> ```Python
>         This is the case if the user is any of the following:
> 
>         * A SALT Astronomer
> ```

For consistency:

A SALT Astronomer ---> a SALT Astronomer
A principal investigator ---> a Principal Investigator
A principal contact ---> a Principal Contact
An administrator --> an administrator
___
*[saltapi/service/permission_service.py, line 125 at r4](https://reviewable.io/reviews/saltastroops/salt-api-2021/28#-Mm-GIunAbqZvUYv718L:-Mm-GIunAbqZvUYv718M:b-6mjg1o) ([raw file](https://github.com/saltastroops/salt-api-2021/blob/639666ff12c60fb6ef9a2a6b00b51eb711f2e378/saltapi/service/permission_service.py#L125)):*
> ```Python
>                or self.user_repository.is_administrator(username)
> 
>     def may_view_comment(self, user: User, proposal_code: ProposalCode):
> ```

may_view_comments ---> may_view_observation_comments

I'd use the plural, as the user may (or may not) view _all_ comments for the proposal.
___
*[saltapi/service/permission_service.py, line 127 at r4](https://reviewable.io/reviews/saltastroops/salt-api-2021/28#-Mm-GBadAXtjdjS_UxPI:-Mm-GBadAXtjdjS_UxPJ:bby4u6x) ([raw file](https://github.com/saltastroops/salt-api-2021/blob/639666ff12c60fb6ef9a2a6b00b51eb711f2e378/saltapi/service/permission_service.py#L127)):*
> ```Python
>     def may_view_comment(self, user: User, proposal_code: ProposalCode):
>         """
>         Checks if the user can update an observation comment
> ```

update an observation comment ---> view the observation comments
___
*[saltapi/service/permission_service.py, line 136 at r4](https://reviewable.io/reviews/saltastroops/salt-api-2021/28#-Mm-GvWeE6TSmUwYMAde:-Mm-GvWeE6TSmUwYMAdf:b-sa0zyb) ([raw file](https://github.com/saltastroops/salt-api-2021/blob/639666ff12c60fb6ef9a2a6b00b51eb711f2e378/saltapi/service/permission_service.py#L136)):*
> ```Python
>         * An administrator
>         """
>         username = user.username
> ```

While this is okay, you could simplify the code by saying that a user may view comments if they may add them.
___
*[saltapi/service/proposal_service.py, line 6 at r3](https://reviewable.io/reviews/saltastroops/salt-api-2021/28#-Mm-Hl9p77h79EdATcPu:-Mm-Hl9p77h79EdATcPv:ba35exf) ([raw file](https://github.com/saltastroops/salt-api-2021/blob/873992b79949b3a20d08fd7128e16f0cf76894f6/saltapi/service/proposal_service.py#L6)):*
> ```Python
> from saltapi.service.proposal import Proposal, ProposalListItem
> from saltapi.service.user import User
> from saltapi.web.schema.proposal import ObservationComment
> ```

ObservationComment shouldn't be used be used in a service, as it is an API level type.
___
*[saltapi/service/proposal_service.py, line 43 at r4](https://reviewable.io/reviews/saltastroops/salt-api-2021/28#-Mm-IDnu1A8rGmPcmB1r:-Mm-IDnu1A8rGmPcmB1s:bs9tw42) ([raw file](https://github.com/saltastroops/salt-api-2021/blob/639666ff12c60fb6ef9a2a6b00b51eb711f2e378/saltapi/service/proposal_service.py#L43)):*
> ```Python
>         return self.repository.get_observation_comments(proposal_code)
> 
>     def add_observation_comment(
> ```

Can we discuss the return value?
___
*[saltapi/web/api/proposals.py, line 378 at r4](https://reviewable.io/reviews/saltastroops/salt-api-2021/28#-Mm-JFF8-D9voE4O6vwT:-Mm-JFF8-D9voE4O6vwU:bnhv7gq) ([raw file](https://github.com/saltastroops/salt-api-2021/blob/639666ff12c60fb6ef9a2a6b00b51eb711f2e378/saltapi/web/api/proposals.py#L378)):*
> ```Python
>     """
>     Adds a new comment related to an observation. The user submitting the request is
>     recorded as the comment user.
> ```

"author" was the correct word here.


<!-- Sent from Reviewable.io -->

hettlage committed:
Merge branch 'development' into observation-comments

Eb-Zeero reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/saltastroops/salt-api-2021/28)* status: 7 of 13 files reviewed, 14 unresolved discussions (waiting on @Eb-Zeero and @hettlage)

___
*[saltapi/repository/proposal_repository.py, line 1021 at r1](https://reviewable.io/reviews/saltastroops/salt-api-2021/28#-MjnpBllDflUvfu8ZlqV:-MmTSgxyAXSUOmvm_oI4:bhgsw8v) ([raw file](https://github.com/saltastroops/salt-api-2021/blob/b503a03e4d1c9064b0010df461ba350bfed8f268/saltapi/repository/proposal_repository.py#L1021)):*
<details><summary><i>Previously, hettlage wrote…</i></summary><blockquote>

But here "author" was correct... ;-)
</blockquote></details>

I need user here to return the author or I am calling the User author.
___
*[saltapi/repository/proposal_repository.py, line 23 at r3](https://reviewable.io/reviews/saltastroops/salt-api-2021/28#-Mm-D_hG9K8i3NhJyOEe:-MmTuO055fFonMz7Y18_:b729yg8) ([raw file](https://github.com/saltastroops/salt-api-2021/blob/873992b79949b3a20d08fd7128e16f0cf76894f6/saltapi/repository/proposal_repository.py#L23)):*
<details><summary><i>Previously, hettlage wrote…</i></summary><blockquote>

As ObservationComment is an API level type, it shouldn't be used in a repository.
</blockquote></details>

we need to discuss this.
___
*[saltapi/service/permission_service.py, line 49 at r4](https://reviewable.io/reviews/saltastroops/salt-api-2021/28#-Mm-Er6o0BnLFBELVNy5:-MmTuTgC61G2KzdkyQm9:b-896fix) ([raw file](https://github.com/saltastroops/salt-api-2021/blob/639666ff12c60fb6ef9a2a6b00b51eb711f2e378/saltapi/service/permission_service.py#L49)):*
<details><summary><i>Previously, hettlage wrote…</i></summary><blockquote>

        This is the case if the user is any of the following:

        * the Principal Investigator (and the proposal can be activated by the PI or PC)
        * the Principal Contact (and the proposal can be activated by the PI or PC)
        * a SALT Astronomer
        * an administrator

</blockquote></details>

Done.
___
*[saltapi/service/permission_service.py, line 108 at r4](https://reviewable.io/reviews/saltastroops/salt-api-2021/28#-Mm-EB3438hZdeOQ-P5W:-MmTriaF9cACJwnf2Fu0:b-896fix) ([raw file](https://github.com/saltastroops/salt-api-2021/blob/639666ff12c60fb6ef9a2a6b00b51eb711f2e378/saltapi/service/permission_service.py#L108)):*
<details><summary><i>Previously, hettlage wrote…</i></summary><blockquote>

Better: may_add_observation_comment (as there are other types of comment).
</blockquote></details>

Done.
___
*[saltapi/service/permission_service.py, line 114 at r4](https://reviewable.io/reviews/saltastroops/salt-api-2021/28#-Mm-FSyH2QsuKPPQ1io-:-MmTrvheAfg8bzU4VOfS:b-896fix) ([raw file](https://github.com/saltastroops/salt-api-2021/blob/639666ff12c60fb6ef9a2a6b00b51eb711f2e378/saltapi/service/permission_service.py#L114)):*
<details><summary><i>Previously, hettlage wrote…</i></summary><blockquote>

For consistency:

A SALT Astronomer ---> a SALT Astronomer
A principal investigator ---> a Principal Investigator
A principal contact ---> a Principal Contact
An administrator --> an administrator
</blockquote></details>

Done.
___
*[saltapi/service/permission_service.py, line 125 at r4](https://reviewable.io/reviews/saltastroops/salt-api-2021/28#-Mm-GIunAbqZvUYv718L:-MmTs5fL2xiE1tn5I3Zn:b-896fix) ([raw file](https://github.com/saltastroops/salt-api-2021/blob/639666ff12c60fb6ef9a2a6b00b51eb711f2e378/saltapi/service/permission_service.py#L125)):*
<details><summary><i>Previously, hettlage wrote…</i></summary><blockquote>

may_view_comments ---> may_view_observation_comments

I'd use the plural, as the user may (or may not) view _all_ comments for the proposal.
</blockquote></details>

Done.
___
*[saltapi/service/permission_service.py, line 127 at r4](https://reviewable.io/reviews/saltastroops/salt-api-2021/28#-Mm-GBadAXtjdjS_UxPI:-MmTsDT-5R0G_8qH0FrN:b-896fix) ([raw file](https://github.com/saltastroops/salt-api-2021/blob/639666ff12c60fb6ef9a2a6b00b51eb711f2e378/saltapi/service/permission_service.py#L127)):*
<details><summary><i>Previously, hettlage wrote…</i></summary><blockquote>

update an observation comment ---> view the observation comments
</blockquote></details>

Done.
___
*[saltapi/service/permission_service.py, line 136 at r4](https://reviewable.io/reviews/saltastroops/salt-api-2021/28#-Mm-GvWeE6TSmUwYMAde:-MmTtvTf3MJ7k3KrdRRw:b-896fix) ([raw file](https://github.com/saltastroops/salt-api-2021/blob/639666ff12c60fb6ef9a2a6b00b51eb711f2e378/saltapi/service/permission_service.py#L136)):*
<details><summary><i>Previously, hettlage wrote…</i></summary><blockquote>

While this is okay, you could simplify the code by saying that a user may view comments if they may add them.
</blockquote></details>

Done.
___
*[saltapi/service/proposal_service.py, line 43 at r4](https://reviewable.io/reviews/saltastroops/salt-api-2021/28#-Mm-IDnu1A8rGmPcmB1r:-MmTu02SCVChG6oxqhSG:b-juksbo) ([raw file](https://github.com/saltastroops/salt-api-2021/blob/639666ff12c60fb6ef9a2a6b00b51eb711f2e378/saltapi/service/proposal_service.py#L43)):*
<details><summary><i>Previously, hettlage wrote…</i></summary><blockquote>

Can we discuss the return value?
</blockquote></details>

yes we can
___
*[saltapi/web/api/proposals.py, line 378 at r4](https://reviewable.io/reviews/saltastroops/salt-api-2021/28#-Mm-JFF8-D9voE4O6vwT:-MmTuHhNE5kRl6KmCff1:b-896fix) ([raw file](https://github.com/saltastroops/salt-api-2021/blob/639666ff12c60fb6ef9a2a6b00b51eb711f2e378/saltapi/web/api/proposals.py#L378)):*
<details><summary><i>Previously, hettlage wrote…</i></summary><blockquote>

"author" was the correct word here.
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

Eb-Zeero committed:
review fix.

Eb-Zeero committed:
Merge remote-tracking branch 'origin/observation-comments' into observation-comments

# Conflicts:
#	saltapi/service/permission_service.py

Eb-Zeero reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/saltastroops/salt-api-2021/28)* status: 7 of 13 files reviewed, 14 unresolved discussions (waiting on @hettlage)

___
*[saltapi/repository/proposal_repository.py, line 1128 at r4](https://reviewable.io/reviews/saltastroops/salt-api-2021/28#-Mm-DxgR9A1rr6uZ329T:-MmYngAt9xp1hUJ3NR3B:b-896fix) ([raw file](https://github.com/saltastroops/salt-api-2021/blob/639666ff12c60fb6ef9a2a6b00b51eb711f2e378/saltapi/repository/proposal_repository.py#L1128)):*
<details><summary><i>Previously, hettlage wrote…</i></summary><blockquote>

Can we discuss the return value?
</blockquote></details>

Done.
___
*[saltapi/service/permission_service.py, line 79 at r4](https://reviewable.io/reviews/saltastroops/salt-api-2021/28#-Mm-FLG32Ig-qLqRoW_5:-MmYnhP_1M3gIKjzIMfW:b-896fix) ([raw file](https://github.com/saltastroops/salt-api-2021/blob/639666ff12c60fb6ef9a2a6b00b51eb711f2e378/saltapi/service/permission_service.py#L79)):*
<details><summary><i>Previously, hettlage wrote…</i></summary><blockquote>

a Principal ---> the Principal
</blockquote></details>

Done.
___
*[saltapi/service/permission_service.py, line 80 at r4](https://reviewable.io/reviews/saltastroops/salt-api-2021/28#-Mm-FQI8-0ARtbPywGqn:-MmYnhoM9XYByItRK599:b-896fix) ([raw file](https://github.com/saltastroops/salt-api-2021/blob/639666ff12c60fb6ef9a2a6b00b51eb711f2e378/saltapi/service/permission_service.py#L80)):*
<details><summary><i>Previously, hettlage wrote…</i></summary><blockquote>

a Principal ---> the Principal
</blockquote></details>

Done.
___
*[saltapi/service/proposal_service.py, line 6 at r3](https://reviewable.io/reviews/saltastroops/salt-api-2021/28#-Mm-Hl9p77h79EdATcPu:-MmYnzV_2VH2ERVTTZdU:b-896fix) ([raw file](https://github.com/saltastroops/salt-api-2021/blob/873992b79949b3a20d08fd7128e16f0cf76894f6/saltapi/service/proposal_service.py#L6)):*
<details><summary><i>Previously, hettlage wrote…</i></summary><blockquote>

ObservationComment shouldn't be used be used in a service, as it is an API level type.
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

Eb-Zeero committed:
review fix 3.

Eb-Zeero committed:
Merge branch 'development' into observation-comments

hettlage committed:
Fixed various issues

hettlage committed:
Fixed various issues

hettlage reviewed (commented):
> Reviewed 4 of 6 files at r5, 4 of 4 files at r8, 2 of 8 files at r9, 1 of 8 files at r10.
> *[Reviewable](https://reviewable.io/reviews/saltastroops/salt-api-2021/28)* status: 8 of 16 files reviewed, all discussions resolved (waiting on @hettlage)



<!-- Sent from Reviewable.io -->

hettlage committed:
Fixed various issues
